user_threshold/models/res_users.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+import os
+from lxml import etree
+
+from odoo import _, api, fields, models
+from odoo.exceptions import AccessError, ValidationError
+
+from .ir_config_parameter import HIDE_THRESHOLD, MAX_DB_USER_PARAM
+from .res_groups import THRESHOLD_MANAGER
+
+
+class ResUsers(models.Model):
+    _inherit = 'res.users'
+
+    threshold_exempt = fields.Boolean(
+        'Exempt User From User Count Thresholds',
+    )
+
+    def __init__(self, pool, cr):
+        """ Override to check if env var to hide threshold configuration and
+        reset the database state is set. If it is, run those actions
+        """
+        if HIDE_THRESHOLD:
+            exempt_users = os.environ.get(
+                'USER_THRESHOLD_USER', ''
+            ).split(',')
+            cr.execute(
+                "SELECT name FROM ir_module_module WHERE "
+                "name='user_threshold' AND state='installed'"
+            )
+            if cr.fetchall():
+                query = """ UPDATE res_users SET threshold_exempt='False'
+                WHERE share='False'"""
+                if exempt_users:
+                    query = "%s AND login NOT IN ('%s')" % (
+                        query, "','".join(exempt_users)
+                    )
+                cr.execute(query)
+
+    def _check_thresholds(self):
+        """ Check to see if any user thresholds are met
+        Returns:
+            False when the thresholds aren't met and True when they are
+        """
+        domain = [
+            ('threshold_exempt', '=', False),
+            ('share', '=', False),
+        ]
+        db_users = len(self.env['res.users'].search(domain))
```

Comment:
I can't use `search_count` if I want to use `.filtered()` later on the same object as the aforementioned returns a long, not a RecordSet